File: LC-Python-Solution/169. Majority Element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:

        numDic = {}

        for num in nums:
            if num in numDic:
                numDic[num] +=1
            else:
                numDic[num] = 1
        
        maxItem = 0
        maxVal = 0

        for item,val in numDic.items():
            if (val > maxVal):
                maxVal = max(maxVal,val)
                maxItem = item
        
        return maxItem


        

File: LC_JS-SOLUTIONS/169. Majority Element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function (nums) {
  let numDic = {};
  for (let num of nums) {
    if (num in numDic) {
      numDic[num] += 1;
    } else {
      numDic[num] = 1;
    }
  }

  let maxItem = 0;
  let maxVal = 0;

  for (const [item, val] of Object.entries(numDic)) {
    if (val > maxVal) {
      maxVal = Math.max(maxVal, val);
      maxItem = item;
    }
  }
  return maxItem;
};
